<launch>

  <param name="use_sim_time" value="false"/>

  <arg name="rviz_config" default="$(find obstacle_detection)/launch/obstacle_view.rviz"/>  
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rviz_config)" />
  
  <node pkg="tf" type="static_transform_publisher" name="map_to_odom" args="0 0 0 0 0 0 map odom 100" />
  <node pkg="tf" type="static_transform_publisher" name="base_to_camera" args="0 0 0 0 0 0 base_link camera_link 100" />
  <node pkg="tf" type="static_transform_publisher" name="base_to_laser" args="0 0 0 0 0 0 base_link laser 100" />
  
  <node pkg="obstacle_detection" type="obstacle_detection" name="obstacle_detection" output="screen">
    <param name="minimum_range" value="0.0"  /> 
    <param name="maximum_range" value="2.0"  /> 
    <param name="voxel_size" value="0.1"  /> 
  </node>

  <include file="$(find depth_image_proc)/launch/pointcloud.launch">
  </include> 

  <include file="$(find depth_image_proc)/launch/pointcloud.launch">
  </include> 

  <include file="$(find amcl)/examples/amcl_diff.launch">
  </include>

  <node pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node" name="pointcloud_to_laserscan">
    <remap from="cloud_in" to="/obstacle_cloud"/>
    <remap from="scan" to="camera_scan"/>
    <rosparam>     
      transform_tolerance: 0.01
      min_height: -1.0
      max_height: 2.0
      angle_increment: 0.0001 # M_PI/360.0
      scan_time: 0.02
      range_min: 0.0
      range_max: 6.0
      use_inf: true
      concurrency_level: 1
    </rosparam>
  </node>

</launch>
